egs/tedlium/s5_r2/local/run_learn_lex.sh

```diff
@@ -0,0 +1,135 @@
+  cat exp/g2p/oov_lex_train/lexicon.lex | awk '{print $1,$3}' | \
+    tr -s '\t' ' ' | sort | uniq > $data/lexicon_oov_g2p.txt || exit 1;
+fi
+
+# Learn a lexicon based on the acoustic training data and the reference lexicon.
+if [ $stage -le 1 ]; then
+  steps/dict/learn_lexicon.sh --g2p-pron-candidates "$data/lexicon_oov_g2p.txt" \
+    --min-prob $min_prob --variants-prob-mass $variants_prob_mass \
+    --variants-prob-mass2 $variants_prob_mass2  \
+    --prior-counts-tot $prior_counts_tot --prior-mean $prior_mean \
+    --affix $affix --stage $lexlearn_stage \
+    --nj 60 --oov-symbol $oov_symbol --retrain-src-mdl false \
+    $ref_dict $ref_vocab $data exp/tri3 data/lang data/local/dict_learned_nosp || exit 1;
+fi
+
+# Add pronounciation probs to the learned lexicon.
+if [ $stage -le 1 ]; then
+  utils/prepare_lang.sh --phone-symbol-table data/lang/phones.txt \
+    data/local/dict_learned_nosp $oov_symbol data/local/lang_learned_nosp data/lang_learned_nosp || exit 1;
+  
+  steps/align_si.sh --nj $nj --cmd "$train_cmd" \
+    $data data/lang_learned_nosp exp/tri2 exp/tri2_ali_learned_lex_nosp || exit 1;
+  
+  steps/get_prons.sh --cmd "$train_cmd" data/train data/lang_learned_nosp exp/tri2_ali_learned_lex_nosp || exit 1;
+  
+  utils/dict_dir_add_pronprobs.sh --max-normalize true \
+    data/local/dict_learned_nosp exp/tri2_ali_learned_lex_nosp/pron_counts_nowb.txt \
+    exp/tri2_ali_learned_lex_nosp/sil_counts_nowb.txt \
+    exp/tri2_ali_learned_lex_nosp/pron_bigram_counts_nowb.txt data/local/dict_learned || exit 1;
+  
+  utils/prepare_lang.sh --phone-symbol-table data/lang/phones.txt \
+    data/local/dict_learned $oov_symbol data/local/lang_learned data/lang_learned || exit 1;
+fi
+
+# Re-train the acoustic model using the learned lexicon
+if [ $stage -le 2 ]; then
+  steps/align_fmllr.sh --nj $nj --cmd "$train_cmd" \
+    $data data/lang_learned exp/tri3 exp/tri3_ali_learned_lex || exit 1;
+  
+  steps/train_sat.sh --cmd "$train_cmd" \
+    5000 100000 $data data/lang_learned exp/tri3_ali_learned_lex exp/tri3_learned_lex || exit 1;
+fi
+
+# Decode
+if [ $stage -le 3 ]; then
+  cp -rT data/lang_learned data/lang_learned_rescore || exit 1;
+  ! cmp data/lang_nosp/words.txt data/lang_learned/words.txt &&\
+    echo "$0: The vocab of the learned lexicon and the reference vocab may be incompatible."
+  cp data/lang_nosp/G.fst data/lang_learned/
+  cp data/lang_nosp_rescore/G.carpa data/lang_learned_rescore/
+  utils/mkgraph.sh data/lang_learned exp/tri3_learned_lex exp/tri3_learned_lex/graph || exit 1;
+  
+  for dset in dev test; do
+  (  steps/decode_fmllr.sh --nj $decode_nj --cmd "$decode_cmd"  --num-threads 4 \
+      exp/tri3_learned_lex/graph data/${dset} exp/tri3_learned_lex/decode_${dset} || exit 1;
+    steps/lmrescore_const_arpa.sh --cmd "$decode_cmd" data/lang_learned data/lang_learned_rescore \
+       data/${dset} exp/tri3_learned_lex/decode_${dset} exp/tri3_learned_lex/decode_${dset}_rescore || exit 1;
+  ) &
+  done
+fi
```

Comment:
Can you put results somewhere in the script, so I can see what the WER is before and after?